src/main/java/io/cdap/plugin/salesforce/SObjectFilterDescriptor.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.salesforce;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * SObject query filter values holder.
+ */
+public class SObjectFilterDescriptor {
+
+  private final ZonedDateTime startDatetime;
+  @Nullable
+  private final String datetimeAfter;
+  @Nullable
+  private final String datetimeBefore;
+  private final Map<ChronoUnit, Integer> duration;
+  private final Map<ChronoUnit, Integer> offset;
+
+  public SObjectFilterDescriptor(@Nullable String datetimeAfter,
+                                 @Nullable String datetimeBefore,
+                                 Map<ChronoUnit, Integer> duration,
+                                 Map<ChronoUnit, Integer> offset) {
+    this(null, datetimeAfter, datetimeBefore, duration, offset);
+  }
+
+  public SObjectFilterDescriptor(@Nullable Long startTime,
+                                 @Nullable String datetimeAfter,
```

Comment:
It's better if this class takes ZonedDateTime instead of the string representations, in order to guarantee that anything that uses this class doesn't have to worry about whether these strings are invalid values.

I also think it's better if there is one constructor for before and after, and one for duration offset:

```
public SObjectFilterDescriptor(@Nullable ZonedDateTime datetimeAfter, @Nullable ZonedDateTime datetimeBefore) {
  ...
}

public SObjectFilterDescriptor(long logicalStartTime, Map<ChronoUnit, Integer> duration, Map<ChronoUnit, Integer> offset) {
  ...
}
```

Reply:
What I meant is that it's better for that logic to be encapsulated in this class instead of in some utility class. Regardless of whether before/after is given, or duration/offset is given, the final query just cares about a start time and an end time. They are different ways of conveying the same information. 

It is cleaner and more re-usable if callers don't need to handle that logic and only need to handle a nullable start time and nullable end time.